pkg/config/validation/validation.go

```diff
@@ -3308,3 +3327,134 @@ func validateNetwork(network *meshconfig.Network) (errs error) {
 func (aae *AnalysisAwareError) Error() string {
 	return aae.Msg
 }
+
+// ValidateTelemetry validates a Telemetry.
+var ValidateTelemetry = registerValidateFunc("ValidateTelemetry",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*telemetry.Telemetry)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateTelemetryMetrics(spec.Metrics),
+			validateTelemetryTracing(spec.Tracing),
+			validateTelemetryAccessLogging(spec.AccessLogging),
+		)
+		return errs.Unwrap()
+	})
+
+func validateTelemetryAccessLogging(logging []*telemetry.AccessLogging) (v Validation) {
+	if len(logging) > 1 {
+		v = appendWarningf(v, "multiple accessLogging is not currently supported")
+	}
+	for idx, l := range logging {
+		if l == nil {
+			continue
+		}
+		if len(l.Providers) > 1 {
+			v = appendValidation(v, Warningf("accessLogging[%d]: multiple providers is not currently supported", idx))
+		}
+		v = appendValidation(v, validateTelemetryProviders(l.Providers))
+	}
+	return
+}
+
+func validateTelemetryTracing(tracing []*telemetry.Tracing) (v Validation) {
+	if len(tracing) > 1 {
+		v = appendWarningf(v, "multiple tracing is not currently supported")
+	}
+	for _, l := range tracing {
+		if l == nil {
+			continue
+		}
+		if len(l.Providers) > 1 {
+			v = appendWarningf(v, "multiple providers is not currently supported")
+		}
+		v = appendValidation(v, validateTelemetryProviders(l.Providers))
+		if l.RandomSamplingPercentage.GetValue() < 0 || l.RandomSamplingPercentage.GetValue() > 100 {
+			v = appendErrorf(v, "randomSamplingPercentage must be in range [0.0, 100.0]")
+		}
+		for name, tag := range l.CustomTags {
+			if name == "" {
+				v = appendErrorf(v, "tag name may not be empty")
+			}
+			switch t := tag.Type.(type) {
+			case *telemetry.Tracing_CustomTag_Literal:
```

Comment:
Any additional restrictions on `CustomTag` value we can validate for? Or all strings make valid tags?

Reply:
I am pretty sure its a CEL expression. It could be nice to validate that, but i am not sure the details on that. @kyessenov @douglas-reid any thoughts?

Reply:
Here, `Literal` represents a static value that gets added to each span. The other two choices are an environment variable and a request header.

Reply:
oh I was thinking of MetricsOverrides_TagOverride.Value which is a CEL expression?

Reply:
yes.

Reply:
We should be parsing all the various languages here (custom targs, access logs, CEL). They are highly error-prone at syntax level. cel-go can be used for validation, but the others one need new parsers in golang.

Reply:
What is the format for custom tags?

Access logs we don't currently support format in the API so we can defer that I think.

This PR is already fairly large so may want to do this in a followup if we need a new library